src/panels/config/automation/target/ha-automation-row-targets.ts

```diff
@@ -0,0 +1,260 @@
+  private devices!: HomeAssistant["devices"];
+
+  @state()
+  @consume({ context: statesContext, subscribe: true })
+  private states!: HomeAssistant["states"];
+
+  @state()
+  @consume({ context: labelsContext, subscribe: true })
+  private _labelRegistry!: LabelRegistryEntry[];
+
+  private _configEntryLookup?: Record<string, ConfigEntry>;
+
+  protected render() {
+    const length = Object.keys(this.target || {}).length;
+    if (!length) {
+      return html`<span class="target">
+        <div class="label">
+          ${this.localize(
+            "ui.panel.config.automation.editor.target_summary.no_target"
+          )}
+        </div>
+      </span>`;
+    }
+    const totalLength = Object.values(this.target || {}).reduce(
+      (acc, val) => acc + ensureArray(val).length,
+      0
+    );
+
+    if (totalLength <= 5) {
+      const targets = Object.entries(this.target!).reduce<
+        ["floor" | "area" | "device" | "entity" | "label", string][]
+      >((acc, [targetType, targetId]) => {
+        const type = targetType.replace("_id", "") as
+          | "floor"
+          | "area"
+          | "device"
+          | "entity"
+          | "label";
+        return [
+          ...acc,
+          ...ensureArray(targetId).map((id): [typeof type, string] => [
+            type,
+            id,
+          ]),
+        ];
+      }, []);
+
+      return targets.map(
+        ([targetType, targetId]) =>
+          html`<span class="target-wrapper">
+            ${this._renderTarget(targetType, targetId)}
+          </span>`
+      );
+    }
+
+    return html`<span class="target">
+      <ha-svg-icon .path=${mdiFormatListBulleted}></ha-svg-icon>
+      <div class="label">
+        ${this.localize(
+          "ui.panel.config.automation.editor.target_summary.targets",
+          {
+            count: totalLength,
+          }
+        )}
+      </div>
+    </span>`;
+  }
+
+  private _getLabel = (id: string) =>
+    this._labelRegistry?.find(({ label_id }) => label_id === id);
+
+  private _checkTargetExists(
+    targetType: "floor" | "area" | "device" | "entity" | "label",
+    targetId: string
+  ): boolean {
+    if (targetType === "floor") {
+      return !!this.floors[targetId];
+    }
+    if (targetType === "area") {
+      return !!this.areas[targetId];
+    }
+    if (targetType === "device") {
+      return !!this.devices[targetId];
+    }
+    if (targetType === "entity") {
+      return !!this.states[targetId];
+    }
+    if (targetType === "label") {
+      return !!this._getLabel(targetId);
+    }
+    return false;
+  }
+
+  private _renderTargetBadge(
+    icon: TemplateResult | typeof nothing,
+    label: string,
+    alert = false
+  ) {
+    return html`<div class="target ${alert ? "alert" : ""}">
+      ${icon}
+      <div class="label">${label}</div>
+    </div>`;
+  }
+
+  private async _loadConfigEntries() {
+    const configEntries = await getConfigEntries(this.hass);
```

Comment:
It's not great that we do this call for every row but maybe we can optimize it later